### PR TITLE
Fix makefile to allow make images to work

### PR DESCRIPTION
**Pull request by soltysh**
/assign @damemi

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,16 +9,15 @@ include $(addprefix ./vendor/github.com/openshift/library-go/alpha-build-machine
...
-# $1 - target suffix
-# $2 - Dockerfile path
-# $3 - context directory for image build
-# It will generate target "image-$(1)" for builing the image an binding it as a prerequisite to target "images".
-$(call build-image,origin-$(GO_PACKAGE),./Dockerfile,.)
+# $1 - target name
+# $2 - image ref
+# $3 - Dockerfile path
+# $4 - context directory for image build
+$(call build-image,ocp-cluster-kube-scheduler-operator,$(IMAGE_REGISTRY)/ocp/4.2:cluster-kube-scheduler-operator, ./Dockerfile.rhel7,.)
```
Comment (ingvagabund) [inline Makefile]: From https://github.com/openshift/cluster-kube-scheduler-operator/blob/master/vendor/github.com/openshift/library-go/alpha-build-machinery/make/targets/openshift/images.mk#L28:

```sh
define build-image
$(eval $(call build-image-internal,$(1),$(2),$(3),$(4)))
endef
```

```sh
# $1 - target name
# $2 - image ref
# $3 - Dockerfile path
# $4 - context
define build-image-internal
```

Which, once rendered, will define Makefile's `image-ocp-cluster-kube-scheduler-operator` target:
```sh
$ make help | grep image
image-ocp-cluster-kube-scheduler-operator
images
```

Once `make images` is run, one will get:
```sh
$ imagebuilder --allow-pull -t registry.svc.ci.openshift.org/ocp/4.2:cluster-kube-scheduler-operator -f ./Dockerfile.rhel7 .
```

Looks good up to the point of running `imagebuilder` command.

Comment (ingvagabund) [reply Makefile]: Nit: `$(IMAGE_REGISTRY)/ocp/4.2:cluster-kube-scheduler-operator` could be put under another variable, e.g.:

```
IMAGE_REGISTRY :=registry.svc.ci.openshift.org
TARGET_OPERATOR_IMAGE :=$(IMAGE_REGISTRY)/ocp/4.2:cluster-kube-scheduler-operator
...
$(call build-image,ocp-cluster-kube-scheduler-operator,$(TARGET_OPERATOR_IMAGE), ./Dockerfile.rhel7,.)
```

to improve readability.

File: Makefile [hunk 2]
```diff
@@ -9,16 +9,15 @@ include $(addprefix ./vendor/github.com/openshift/library-go/alpha-build-machine
 	targets/openshift/images.mk \
 )
 
-# Exclude e2e tests from unit testing
-GO_TEST_PACKAGES :=./pkg/... ./cmd/...
```
Comment (tnozicka) [inline Makefile]: global vars are by convention at the start of the file grouped together

File: Makefile (hunk 1 above)
Comment (tnozicka) [reply Makefile]: I think it less complicated without that extra variable. variables are intended to be set from outside - you can't change component name, just the registry.

let's take https://github.com/openshift/oc/blob/master/Makefile#L45 as a reference and keep it mostly similar

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,8 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-kube-scheduler-operator
 COPY . .
-RUN go build ./cmd/cluster-kube-scheduler-operator
+ENV GO_PACKAGE github.com/openshift/cluster-kube-scheduler-operator
```
Comment (tnozicka) [inline Dockerfile]: drop it, autodetected

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
+RUN go build -ldflags "-X $GO_PACKAGE/pkg/version.versionFromGit=$(git describe --long --tags --abbrev=7 --match 'v[0-9]*')" ./cmd/cluster-kube-scheduler-operator
```
Comment (tnozicka) [inline Dockerfile]: make build

File: Dockerfile.rhel7 [hunk 5]
```diff
@@ -1,7 +1,8 @@
 FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.12 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-kube-scheduler-operator
 COPY . .
-RUN go build ./cmd/cluster-kube-scheduler-operator
+ENV GO_PACKAGE github.com/openshift/cluster-kube-scheduler-operator
+RUN go build -ldflags "-X $GO_PACKAGE/pkg/version.versionFromGit=$(git describe --long --tags --abbrev=7 --match 'v[0-9]*')" ./cmd/cluster-kube-scheduler-operator
```
Comment (tnozicka) [inline Dockerfile.rhel7]: make build

File: Dockerfile.rhel7 [hunk 6]
```diff
@@ -1,7 +1,8 @@
 FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.12 AS builder
```
Comment (tnozicka) [inline Dockerfile.rhel7]: i think this is technically .rhel8 as that's the base image is comes from, admiting it's preexisting

Comment (soltysh) [reply Dockerfile.rhel7]: :man_shrugging: I'm leaving this consistent with all the rest.

File: Dockerfile (hunk 4 above)
Comment (tnozicka) [reply Dockerfile]: this was just a hack and doesn't support ART builds

**Pull request merged**